src/systems/capture-system.js

```diff
@@ -0,0 +1,64 @@
+export class CaptureSystem {
+  constructor(scene) {
+    this.scene = scene;
+
+    this.gotAudioTrack = false;
+    this.stream = new MediaStream();
+
+    this.scene.addEventListener("loaded", () => {
+      const video = this.scene.canvas.captureStream().getVideoTracks()[0];
+      this.stream.addTrack(video);
+
+      this.recorder = new MediaRecorder(this.stream);
+
+      this.chunks = [];
+
+      this.recorder.ondataavailable = e => {
+        this.chunks.push(e.data);
+      };
+      this.recorder.onstop = () => {
+        if (this.chunks.length === 0) return;
+        const mimeType = this.chunks[0].type;
+        const blob = new Blob(this.chunks, { type: mimeType });
+        this.chunks.length = 0;
+        this.scene.emit("add_media", new File([blob], "capture", { type: mimeType }));
+      };
+    });
```

Comment:
A `"loaded"` event should not reach the scene more than once, but do you want to add, `{once: true}` to `addEventListener` anyway?